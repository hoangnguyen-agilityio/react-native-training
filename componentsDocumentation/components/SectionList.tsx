import React from 'react';
import {StyleSheet, Text, View, SectionList} from 'react-native';

const DATA = [
  {
    title: 'Main dishes',
    data: ['Pizza', 'Burger', 'Risotto'],
  },
  {
    title: 'Sides',
    data: ['French Fries', 'Onion Rings', 'Fried Shrimps'],
  },
  {
    title: 'Drinks',
    data: ['Water', 'Coke', 'Beer'],
  },
  {
    title: 'Desserts',
    data: ['Cheese Cake', 'Ice Cream'],
  },
];

const Item = ({title}: {title: string}) => (
  <View style={styles.item}>
    <Text style={styles.title}>{title}</Text>
  </View>
);

const Sectionlist = () => (
  <SectionList
    sections={DATA}
    keyExtractor={(item, index) => item + index}
    renderItem={({item}) => <Item title={item} />}
    renderSectionHeader={({section: {title}}) => (
      <Text style={styles.header} data-testID="section-header">{title}</Text>
    )}
  />
);

const styles = StyleSheet.create({
  item: {
    backgroundColor: '#f9c2ff',
    padding: 20,
    marginVertical: 8,
  },
  header: {
    fontSize: 32,
    backgroundColor: '#fff',
  },
  title: {
    fontSize: 24,
  },
});

export default Sectionlist;
