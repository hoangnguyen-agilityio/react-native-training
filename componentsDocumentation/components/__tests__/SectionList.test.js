import React from 'react';
import renderer from 'react-test-renderer';
import SectionList from '../SectionList';
import { TouchableOpacity } from 'react-native';

test('renders correctly', () => {
  const tree = renderer.create(<SectionList />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('there are 4 sections in the list', () => {
  const tree = renderer.create(<SectionList />);
  const sections = tree.root.findAllByProps({
    ['data-testID']: 'section-header',
  });

  expect(sections).toHaveLength(4);
});
