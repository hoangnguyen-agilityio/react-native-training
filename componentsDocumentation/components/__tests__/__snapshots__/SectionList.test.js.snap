// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "data": Array [
          "Pizza",
          "Burger",
          "Risotto",
        ],
        "title": "Main dishes",
      },
      Object {
        "data": Array [
          "French Fries",
          "Onion Rings",
          "Fried Shrimps",
        ],
        "title": "Sides",
      },
      Object {
        "data": Array [
          "Water",
          "Coke",
          "Beer",
        ],
        "title": "Drinks",
      },
      Object {
        "data": Array [
          "Cheese Cake",
          "Ice Cream",
        ],
        "title": "Desserts",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
      5,
    ]
  }
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        data-testID="section-header"
        style={
          Object {
            "backgroundColor": "#fff",
            "fontSize": 32,
          }
        }
      >
        Main dishes
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Pizza
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Burger
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Risotto
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    >
      <Text
        data-testID="section-header"
        style={
          Object {
            "backgroundColor": "#fff",
            "fontSize": 32,
          }
        }
      >
        Sides
      </Text>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          French Fries
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Onion Rings
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Object {
            "backgroundColor": "#f9c2ff",
            "marginVertical": 8,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
            }
          }
        >
          Fried Shrimps
        </Text>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      style={
        Object {
          "height": 0,
        }
      }
    />
  </View>
</RCTScrollView>
`;
